playwright/tests/pub_yearFilter.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { Given, Then, When } from "@cucumber/cucumber";
+import { expect } from "@playwright/test";
+import { CustomWorld } from "../support/setup";
+
+Given(
+  "I want to filter by publication year",
+  async function (this: CustomWorld) {
+    return await this.page.goto(`${this.parameters.appUrl}`);
+  }
+);
+
+When(
+  "I search for New York to test publication year filter",
+  async function (this: CustomWorld) {
+    await this.page.locator('[aria-label="Item Search"]').fill("New York");
+    await this.page.locator("#searchbar-button-search-bar").click();
```

Comment:
Is there a locator here we could use that isn't an id? Like a button within a form that includes the word "Search"?

Reply:
should be `this.page.getByRole(...)`

we have to use `this.<playwright_api_element>` because we are wrapping playwright in Cucumber which is happening in the `setup.ts` file.

Reply:
sorry, typo from me.  I actually am using 'this'.
![image](https://user-images.githubusercontent.com/1929170/210435375-1b5a3a1c-ce7e-44fe-b6f5-65e8a6eeaf3f.png)
await this.page.getByRole("button", { name: "Search" }).click();

Reply:
interesting how vscode spots the error even before I run it.

Reply:
I'm wondering if there is an additional library import that needed since 'getByRole' is foreign to vscode.

Reply:
This is happening because we need to upgrade Playwright to version 1.29.1. I attempted this but ran into other issues when trying to upgrade. For now, we will leave this locator as is and I'll create another ticket to work on updating Playwright.